### ISSUE TITLE
Action Proposal: Lock/Unlock Semaphore

### ISSUE THREAD

**Issue by kraigparkinson** [sindresorhus/Actions#103]
### Description

I'd like to be able to lock a shortcut semaphore so that I can hold the execution of a shortcut if another instance of the same shortcut (or another shortcut under the same semaphore) is already running. 

Ideally, it has customizable properties like timeout and retry interval. 

When timeout or completion occurs, I'd like it to return a value to see if it completed naturally or timed out, so I set up an if/then that can take appropriate action for the situation.

**Comment by sindresorhus** [comment]
Good idea. On iOS, an action times out after about 30 seconds (enforced by iOS), so this would not work for long-running shortcuts.

**Comment by sindresorhus** [comment]
This was added in the latest TestFlight version of Actions. It would be great if you could help test it out. It won't be out in the App Store for a while. It's called `Manage Shortcut Lock`.

https://testflight.apple.com/join/fJGUrsZx
